components/dashboard/src/components/BillingAccountSelector.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Team } from "@gitpod/gitpod-protocol";
+import { useContext, useEffect, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { TeamsContext } from "../teams/teams-context";
+import { UserContext } from "../user-context";
+import SelectableCardSolid from "../components/SelectableCardSolid";
+import { ReactComponent as Spinner } from "../icons/Spinner.svg";
+
+export function BillingAccountSelector(props: { onSelected?: () => void }) {
+    const { user, setUser } = useContext(UserContext);
+    const { teams } = useContext(TeamsContext);
+    const [teamsWithBillingEnabled, setTeamsWithBillingEnabled] = useState<Team[] | undefined>();
```

Comment:
@jankeromnes, I bet this is not the single place, where we'd need this kind of info. have you considered promoting this flag to the result of getTeams? `team.isBillingEnabled` would be nice and it shave off the additional requests.

Reply:
@AlexTugarev I think this would be premature, because:
- I can't imagine other places that will need the same information currently (note: this billing account selector component is already reusable, and in fact it's already used in two places)
- I don't think it's wise to make `getTeams` load all the team subscriptions on every call, even when most callers don't need that currently